src/spotify-service.ts

```diff
@@ -1,20 +1,35 @@
 import { SpotifyApi, type AccessToken } from "@spotify/web-api-ts-sdk";
-import { Context, Effect, Layer, Secret, Encoding, pipe, Queue } from "effect";
-import { SpotifyConfigService } from "./spotify-config-service";
+import { Context, Data, Effect, Encoding, Layer, Queue, Secret } from "effect";
 import { Message, MessagePubSub } from "./message-pubsub";
+import { SpotifyConfigService } from "./spotify-config-service";
+
+export class SpotifyError extends Data.TaggedError("SpotifyError")<{
+  cause: unknown;
+}> {}
+
+const make = Effect.gen(function* () {
+  const config = yield* SpotifyConfigService;
+
+  const client = SpotifyApi.withAccessToken(
+    config.clientId,
+    config.accessToken
+  );
+  const use = <A>(f: (client: SpotifyApi) => Promise<A>) =>
```

Comment:
One way to wrap Promise apis, is to create a client wrapper that adds a custom error type.

Reply:
Nice, I dig it.